### PR TITLE
Added initial test

### PR DESCRIPTION
**Pull request by tophj-ibm**
Added initial tests. Need to add more.

Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by tophj-ibm** [comment]
don't accept this, I need to clean some things up

File: cli/command/manifest/annotate_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	}{
+		{
+			name:		"wrong-args",
+			args:		[]string{},
+			expectedError:	"\"annotate\" requires exactly 2 argument(s).",
+		},
+	}
+	for _, tc := range testCases {
+		buf := new(bytes.Buffer)
+		cmd := newAnnotateCommand(test.NewFakeCli(&fakeClient{}, buf))
```
Comment (dnephin) [inline cli/command/manifest/annotate_test.go:29]: Just a heads-up that this has changed on master. `NewFakeCli()` no longer takes a second arg. It uses a `bytes.Buffer` by default now, which you can access using the `OutBuffer()` method.

**Comment by dnephin** [comment]
I also noticed this PR is open against `manifest-cmd-history` instead of `manifest-cmd` (which is the branch in https://github.com/docker/cli/pull/138).  Is that expected?

**Comment by clnperez** [comment]
@dnephin history's just a non-squashed version of the other, so at the time they were identical (which they almost always are) and it doesn't matter. 

For these tests, I've been looking more at fakeClient, and it doesn't look like that will test our code since there is no client for the manifest code. That explains why @tophj-ibm  was still seeing things go out to an actual registry.

**Comment by dnephin** [comment]
Yes, that sounds right. The `fakeClient` is just a fake for the docker engine API. There should be a client in here somewhere for the registry as well, right? So we just need to make sure we have an interface for it, then we can use a fake registry client for testing.

**Comment by dnephin** [comment]
I took another look .The client seems to be `distribution.ManifestService`. Since the client depends on the image name/tag it's not something we can pass in to fake, so we'll probably have to do it differently.

I think we can provide our own internal interface for the two requests that are performed (get and push). That way the rest of the code can use a fake.

**Comment by dnephin** [comment]
I can look into that refactor and open another PR for it

**Comment by tophj-ibm** [comment]
closing in favor of #5

**Pull request closed without merging**